web/src/drafts.ts

```diff
@@ -20,45 +21,71 @@ import * as timerender from "./timerender";
 import * as ui_util from "./ui_util";
 import * as util from "./util";
 
-export function set_count(count) {
+export function set_count(count: number): void {
     const $drafts_li = $(".top_left_drafts");
     ui_util.update_unread_count_in_dom($drafts_li, count);
 }
 
-export const draft_model = (function () {
-    const exports = {};
+type LocalStorageDraft = {
+    content: string;
+    updatedAt?: number;
+} & (
+    | {
+          type: false;
+      }
+    | {
+          type: "stream";
+          topic: string;
+          stream_id?: number;
+      }
+    | {
+          type: "private";
+          reply_to: string;
+          private_message_recipient: string;
+      }
+);
+
+// The id is added to the draft in format_drafts in drafts_overlay_ui.
+// We should probably just include it in the draft object itself always?
+type LocalStorageDraftWithId = LocalStorageDraft & {id: string};
 
+export const draft_model = (function () {
     // the key that the drafts are stored under.
     const KEY = "drafts";
     const ls = localstorage();
+
+    // TODO(evy) -- this is causing a type error
+    // and I don't see it used anywhere, can we remove it?
     ls.version = 1;
```

Comment:
Question here!

Reply:
Yeah this can be deleted; it was present since 1929cc5190387eab320966f067a78fe1cd728df7 and clearly is intended to be part of a never-implemented format conversion system. We can add a version if/when we actually need such a feature.

Reply:
Great, removed it